backend/de.metas.salescandidate.base/src/main/java/de/metas/ordercandidate/api/source/EligibleOLCandProvider.java

```diff
@@ -0,0 +1,178 @@
+/*
+ * #%L
+ * de.metas.salescandidate.base
+ * %%
+ * Copyright (C) 2022 metas GmbH
+ * %%
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License as
+ * published by the Free Software Foundation, either version 2 of the
+ * License, or (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public
+ * License along with this program. If not, see
+ * <http://www.gnu.org/licenses/gpl-2.0.html>.
+ * #L%
+ */
+
+package de.metas.ordercandidate.api.source;
+
+import com.google.common.collect.ImmutableList;
+import de.metas.async.AsyncBatchId;
+import de.metas.common.util.time.SystemTime;
+import de.metas.impex.InputDataSourceId;
+import de.metas.impex.api.IInputDataSourceDAO;
+import de.metas.logging.LogManager;
+import de.metas.ordercandidate.OrderCandidate_Constants;
+import de.metas.ordercandidate.api.IOLCandBL;
+import de.metas.ordercandidate.api.OLCand;
+import de.metas.ordercandidate.api.OLCandAggregation;
+import de.metas.ordercandidate.api.OLCandAggregationColumn;
+import de.metas.ordercandidate.model.I_C_OLCand;
+import de.metas.util.Services;
+import lombok.NonNull;
+import org.adempiere.ad.dao.IQueryBL;
+import org.slf4j.Logger;
+import org.springframework.stereotype.Service;
+
+import javax.annotation.Nullable;
+import java.time.LocalDate;
+import java.util.List;
+import java.util.Objects;
+
+@Service
+public class EligibleOLCandProvider
+{
+	private static final Logger logger = LogManager.getLogger(EligibleOLCandProvider.class);
+
+	private final IQueryBL queryBL = Services.get(IQueryBL.class);
+	private final IInputDataSourceDAO inputDataSourceDAO = Services.get(IInputDataSourceDAO.class);
+
+	private final LocalDate defaultDateDoc = SystemTime.asLocalDate();
+
+	@NonNull
+	public List<OLCand> getEligibleOLCand(@NonNull final GetEligibleOLCandRequest request)
+	{
+		final InputDataSourceId processorDataDestinationId = inputDataSourceDAO
+				.retrieveInputDataSourceIdByInternalName(OrderCandidate_Constants.DATA_DESTINATION_INTERNAL_NAME);
+
+		final IOLCandBL olCandBL = Services.get(IOLCandBL.class);
+
+		return queryBL.createQueryBuilder(I_C_OLCand.class)
```

Comment:
whats the max size here? i'm concerned it might be too large for the heap.
not sure how it worked in the older code.
suggestion: 
- add the `request.getAggregationInfo()` to the query.
- iterate or iterateAndStream, but don't load it all into a list
- let this method return a stream or an iterator

Reply:
change done accodingly
> iterate or iterateAndStream, but don't load it all into a list